src/Illuminate/Database/Query/Builder.php

```diff
@@ -1395,17 +1395,19 @@ protected function runSelect()
      * @param  int  $perPage
      * @param  array  $columns
      * @param  string  $pageName
+     * @param  int|null  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
      */
-    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page')
+    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page', $page = null)
     {
-        $page = Paginator::resolveCurrentPage($pageName);
```

Comment:
This is the only line within this function that should be changed. Nothing else needs changin.